faust/stores/bigtable.py

```diff
@@ -0,0 +1,605 @@
+                of a changelog event.
+        """
+        tp_offsets: Dict[TP, int] = {}
+        for event in batch:
+            tp, offset = event.message.tp, event.message.offset
+            tp_offsets[tp] = (
+                offset if tp not in tp_offsets else max(offset, tp_offsets[tp])
+            )
+            msg = event.message
+            bt_key = self._add_partition_prefix_to_key(msg.key, msg.partition)
+
+            if msg.value is None:
+                self._del_cache(msg.key)
+                self._bigtable_del(bt_key)
+            else:
+                self._set_cache(msg.key, msg.value)
+                self._bigtable_set(bt_key, msg.value)
+
+        for tp, offset in tp_offsets.items():
+            self.set_persisted_offset(tp, offset)
+
+    async def backup_partition(
+        self,
+        tp: Union[TP, int],
+        flush: bool = True,
+        purge: bool = False,
+        keep: int = 1,
+    ) -> None:
+        """Backup partition from this store.
+
+        Not yet implemented for Bigtable.
+
+        """
+        raise NotImplementedError("Not yet implemented for Bigtable.")
+
+    def restore_backup(
+        self, tp: Union[TP, int], latest: bool = True, backup_id: int = 0
+    ) -> None:
+        """Restore partition backup from this store.
+
+        Not yet implemented for Bigtable.
+
+        """
+        raise NotImplementedError("Not yet implemented for Bigtable.")
+
+    def _fill_caches(self, partitions):
+        for k, v in self._bigtable_iteritems(partitions=partitions):
+            self._set_cache(k, v)
+
+        self._startup_cache_partitions |= set(partitions)
+        # Invalidate startup cache after 30 minutes
+        # or reset the timer if already running
+        if self._invalidation_timer is not None:
+            self._invalidation_timer.cancel()
+            del self._invalidation_timer
+            self._invalidation_timer = None
+        self._invalidation_timer = threading.Timer(
+            self._startup_cache_ttl, self._invalidate_startup_cache
+        )
+        self._invalidation_timer.start()
+
+    def _get_active_changelogtopic_partitions(
+        self, table: CollectionT, tps: Set[TP]
+    ) -> Set[int]:
+        partitions = set()
+        standby_tps = self.app.assignor.assigned_standbys()
+        my_topics = table.changelog_topic.topics
+        for tp in tps:
+            if tp.topic in my_topics and tp not in standby_tps:
+                partitions.add(tp.partition)
+        return partitions
+
+    async def assign_partitions(
+        self, table: CollectionT, tps: Set[TP], generation_id: int = 0
+    ) -> None:
+        # Fill cache with all keys for the partitions we are assigned
+        if self._startup_cache_enable is False:
+            return
+
+        partitions = self._get_active_changelogtopic_partitions(table, tps)
+        if len(partitions) == 0:
+            return
+        self.log.info(f"Assigning partitions {partitions} for {table.name}")
```

Comment:
I think this log should also be logged when the `self._startup_cache_enable == False`